tree/treeplayer/src/TTreeIndex.cxx

```diff
@@ -177,8 +177,19 @@ TTreeIndex::TTreeIndex(const TTree *T, const char *majorname, const char *minorn
          fMajorFormula->UpdateFormulaLeaves();
          fMinorFormula->UpdateFormulaLeaves();
       }
-      tmp_major[i] = (Long64_t) fMajorFormula->EvalInstance<LongDouble_t>();
-      tmp_minor[i] = (Long64_t) fMinorFormula->EvalInstance<LongDouble_t>();
+      auto GetAndRangeCheck = [this](bool isMajor, Long64_t entry) {
+         LongDouble_t ldRet = (isMajor ? fMajorFormula : fMinorFormula)->EvalInstance<LongDouble_t>();
+         Long64_t ret = (Long64_t) ldRet;
+         Long64_t retCloserToZero = ret > 0 ? ret - 1 : ret + 1;
```

Comment:
As is, it will never trigger an error because 'ret' is Long64_t and can not saturate.
```suggestion
         Long64_t retCloserToZero = ret > 0 ? (Long64_t) (ldRet - 1) : (Long64_t) (ldRet + 1);
```